### PR TITLE
feat(infocols-previews): IS-513: Implement infocols previews

### PR DESCRIPTION
**Pull request by harishv7**
## Problem

Infocols has no preview yet

Closes IS-513

## Solution

Implements template previews for infocols on CMS

Screenshot 2023-10-16 at 11 40 36 PM

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [X] No - this PR is backwards compatible with ALL of the following feature flags in this [doc](https://www.notion.so/opengov/Existing-feature-flags-518ad2cdc325420893a105e88c432be5)

### CODE REVIEW

File: src/templates/homepage/InfocolsSection.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      title,
+      subtitle,
+      linktext,
+      url,
+      infoboxes,
+      sectionIndex,
+    }: TemplateTextCardsSectionProps,
+    ref: ForwardedRef<HTMLDivElement>
+  ) => {
+    const getInfoBox = (infoboxTitle: string, infoboxDescription?: string) => {
```
Comment (harishv7) [inline src/templates/homepage/InfocolsSection.tsx:29]: follows same structure as in templates. infobox is an includes file in templates which has been mimicked here with a extracted function

Comment (seaerchin) [reply src/templates/homepage/InfocolsSection.tsx:29]: not a real issue, just curious why you chose to use a function over a component

File: src/layouts/EditHomepage/EditHomepage.jsx [hunk 2]
```diff
@@ -373,23 +374,23 @@ const EditHomepage = ({ match }) => {
             textCardItemErrors.push(
               _.map(cards, () => getErrorValues(TEXTCARDS_ITEM_SECTION))
             )
-            if (textCardItemErrors.length === 0) {
-              textCardItemErrors.push([])
-            }
```
Comment (seaerchin) [inline src/layouts/EditHomepage/EditHomepage.jsx:378]: what happens now when there are no items?

File: src/layouts/EditHomepage/EditHomepage.jsx [hunk 3]
```diff
@@ -373,23 +374,23 @@ const EditHomepage = ({ match }) => {
...
             infocolInfoboxErrors.push(
               _.map(infoboxes, () => getErrorValues(INFOCOLS_INFOBOX_SECTION))
             )
-            if (infocolInfoboxErrors.length === 0) {
-              infocolInfoboxErrors.push([])
-            }
+          } else {
+            // populate for nested components
+            textCardItemErrors.push([])
+            infocolInfoboxErrors.push([])
```
Comment (seaerchin) [inline src/layouts/EditHomepage/EditHomepage.jsx:393]: do we not need to do the checks here? (length === 0)

File: src/layouts/EditHomepage/HomepagePreview.tsx [hunk 4]
```diff
@@ -200,12 +201,16 @@ export const HomepagePreview = ({
...
           {EditorHomepageFrontmatterSection.isInfocols(section) && (
             <>
-              <div
+              <TemplateInfocolsSection
                 key={`section-${sectionIndex}`}
+                title={section.infocols.title}
+                subtitle={section.infocols.subtitle}
+                linktext={section.infocols.linktext}
+                url={section.infocols.url}
+                infoboxes={section.infocols.infoboxes}
```
Comment (seaerchin) [inline src/layouts/EditHomepage/HomepagePreview.tsx:210]: not a blocker, wouldn't using a spread over `section.infocols` be better here?

File: src/styles/isomer-template/components/homepage/_infocols.scss
Comment (seaerchin) [inline src/styles/isomer-template/components/homepage/_infocols.scss:1]: skimmed css review, leaving this to design to check for parity

**Review by seaerchin — APPROVED** [review]
lgtm

File: src/layouts/EditHomepage/EditHomepage.jsx (hunk 3 above)
Comment (harishv7) [reply src/layouts/EditHomepage/EditHomepage.jsx:420]: the length checks were never there. I added it in the parent PR (IS-512) which was wrong. This logic exists to keep track of the errors with the indices representing the section

File: src/layouts/EditHomepage/EditHomepage.jsx (hunk 2 above)
Comment (harishv7) [reply src/layouts/EditHomepage/EditHomepage.jsx:405]: Replied below - this logic is to basically maintain sectionIdx - errors mapping

File: src/layouts/EditHomepage/HomepagePreview.tsx (hunk 4 above)
Comment (harishv7) [reply src/layouts/EditHomepage/HomepagePreview.tsx:210]: true, was following the same format as the above components. updated now

File: src/templates/homepage/InfocolsSection.tsx (hunk 1 above)
Comment (seaerchin) [reply src/templates/homepage/InfocolsSection.tsx:29]: Oh ps, I was ok with the function, just curious on the reason!!! Don't feel pressured to do this

Comment (harishv7) [reply src/templates/homepage/InfocolsSection.tsx]: done

File: src/layouts/EditHomepage/HomepagePreview.tsx (hunk 4 above)
Comment (harishv7) [reply src/layouts/EditHomepage/HomepagePreview.tsx]: done

**Pull request merged**